src/app/public/modules/repeater/repeater.component.ts

```diff
@@ -33,36 +47,62 @@ export class SkyRepeaterComponent implements AfterContentInit {
   @ContentChildren(SkyRepeaterItemComponent)
   public items: QueryList<SkyRepeaterItemComponent>;
 
+  private ngUnsubscribe = new Subject<void>();
+
   private _expandMode = 'none';
 
-  constructor(private repeaterService: SkyRepeaterService) {
-    this.repeaterService.itemCollapseStateChange.subscribe((item: SkyRepeaterItemComponent) => {
-      if (this.expandMode === 'single' && item.isExpanded) {
-        this.items.forEach((otherItem) => {
-          if (otherItem !== item && otherItem.isExpanded) {
-            otherItem.isExpanded = false;
-          }
-        });
-      }
-    });
+  constructor(
+    private repeaterService: SkyRepeaterService
+  ) {
+    this.repeaterService.itemCollapseStateChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((item: SkyRepeaterItemComponent) => {
+        if (this.expandMode === 'single' && item.isExpanded) {
+          this.items.forEach((otherItem) => {
+            if (otherItem !== item && otherItem.isExpanded) {
+              otherItem.isExpanded = false;
+            }
+          });
+        }
+      });
 
     this.updateForExpandMode();
   }
 
   public ngAfterContentInit() {
+    // If activeIndex has been set on init, call service to activate the appropriate item.
+    setTimeout(() => {
+      if (this.activeIndex || this.activeIndex === 0) {
+        this.repeaterService.activateItemByIndex(this.activeIndex);
+      }
+    });
+
     // HACK: Not updating for expand mode in a timeout causes an error.
     // https://github.com/angular/angular/issues/6005
-    this.items.changes.subscribe(() => {
-      setTimeout(() => {
-        this.updateForExpandMode(this.items.last);
-      }, 0);
-    });
+    this.items.changes
+      .takeUntil(this.ngUnsubscribe)
```

Comment:
Added an takeUntil here ☝️for better memory management. Logic should not be changed, as this was just code cleanup unrelated to original issue.